sslcontext-kickstart-for-pem/src/main/java/nl/altindag/ssl/util/PemUtils.java

```diff
@@ -88,8 +88,15 @@
     private static final JcaX509CertificateConverter CERTIFICATE_CONVERTER = new JcaX509CertificateConverter().setProvider(BOUNCY_CASTLE_PROVIDER);
     private static final JceOpenSSLPKCS8DecryptorProviderBuilder OPEN_SSL_PKCS8_DECRYPTOR_PROVIDER_BUILDER = new JceOpenSSLPKCS8DecryptorProviderBuilder().setProvider(BOUNCY_CASTLE_PROVIDER);
     private static final JcePEMDecryptorProviderBuilder PEM_DECRYPTOR_PROVIDER_BUILDER = new JcePEMDecryptorProviderBuilder().setProvider(BOUNCY_CASTLE_PROVIDER);
-    private static final BouncyFunction<char[], InputDecryptorProvider> INPUT_DECRYPTOR_PROVIDER = password -> OPEN_SSL_PKCS8_DECRYPTOR_PROVIDER_BUILDER.build(Objects.requireNonNull(password));
-    private static final BouncyFunction<char[], PEMDecryptorProvider> PEM_DECRYPTOR_PROVIDER = password -> PEM_DECRYPTOR_PROVIDER_BUILDER.build(Objects.requireNonNull(password));
+    private static final BouncyFunction<char[], InputDecryptorProvider> INPUT_DECRYPTOR_PROVIDER = password -> OPEN_SSL_PKCS8_DECRYPTOR_PROVIDER_BUILDER.build(requirePassword(password));
+    private static final BouncyFunction<char[], PEMDecryptorProvider> PEM_DECRYPTOR_PROVIDER = password -> PEM_DECRYPTOR_PROVIDER_BUILDER.build(requirePassword(password));
+
+    private static char[] requirePassword(char[] maybeNull) {
```

Comment:
I think this code snippet can also be used elsewhere. Would it be possible for you to move it into a separate class within the core library? Maybe a class named `ValidationUtils` within the utils package containing this method but slightly adjusted so it can be used for generic purposes? I thought of something like 
```
public static <T> T requireNonNull(T object, Supplier<? extends RuntimeException> exception) {
        ...
}
```

In this way you would be able to call this method on line number 92 like 
```
password -> PEM_DECRYPTOR_PROVIDER_BUILDER.build(ValidationUtils.requireNonNull(password, () -> new IllegalArgumentException("A password is mandatory with an encrypted key")));
```

Reply:
Fair, can do.

I did not do it as today there is a single use case, and I orientate method name towards it.

Reply:
Yes makes totally sense, thank you for refactoring it!